beater/handlers.go

```diff
@@ -87,11 +88,37 @@ func newMuxer(config *Config, report reporter) *http.ServeMux {
 	return mux
 }
 
+func concurrencyLimitHandler(config *Config, h http.Handler) http.Handler {
+	semaphore := make(chan struct{}, config.ConcurrentRequests)
+
+	release := func() {
+		<-semaphore
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		select {
+		case semaphore <- struct{}{}:
+			defer release()
+			h.ServeHTTP(w, r)
+		case <-time.After(config.MaxRequestQueueTime):
+			logger, ok := r.Context().Value(reqLoggerContextKey).(*logp.Logger)
+			if ok {
+				logger.Error("request rejected due to too many concurrent requests")
+			} else {
+				logp.NewLogger("request").Errorf("request rejected due to too many concurrent requests")
```

Comment:
can we flip this to assigning `logger` and consolidate the `logger.Error("request rejected due to too many concurrent requests")` to after the `if` to reduce repetition?

Reply:
Got rid of all of this. WDYT?

Reply:
I like it, didn't realize `sendStatus` existed